lib/error.js

```diff
@@ -539,15 +562,60 @@ AppError.emailBouncedSoft = () => {
   })
 }
 
-AppError.messageRejected = (reason, reasonCode) => {
+AppError.emailExists = () => {
   return new AppError({
-    code: 500,
+    code: 400,
     error: 'Bad Request',
-    errno: ERRNO.MESSAGE_REJECTED,
-    message: 'Message rejected'
-  }, {
-    reason,
-    reasonCode
+    errno: ERRNO.EMAIL_EXISTS,
+    message: 'Email already exists'
+  })
+}
+
+AppError.cannotDeletePrimaryEmail = () => {
+  return new AppError({
+    code: 400,
+    error: 'Bad Request',
+    errno: ERRNO.EMAIL_DELETE_PRIMARY,
+    message: 'Can not delete primary email'
+  })
+}
+
+AppError.unverifiedSession = function () {
+  return new AppError({
+    code: 400,
+    error: 'Bad Request',
+    errno: ERRNO.SESSION_UNVERIFIED,
+    message: 'Unverified session'
+  })
+}
+
+AppError.yourPrimaryEmailExists = () => {
+  return new AppError({
+    code: 400,
+    error: 'Bad Request',
+    errno: ERRNO.USER_PRIMARY_EMAIL_EXISTS,
+    message: 'Can not add secondary email that is same as your primary'
+  })
+}
+
+AppError.verifiedPrimaryEmailAlreadyExists = () => {
+  return new AppError({
+    code: 400,
+    error: 'Bad Request',
+    errno: ERRNO.VERIFIED_PRIMARY_EMAIL_EXISTS,
+    message: 'Email already exists'
+  })
+}
+
+// This error is thrown when someone attempts to add a secondary email
+// that is the same as the primary email of another account, but the account
+// was recently created ( < 24hrs).
+AppError.unverifiedPrimaryEmailNewlyCreated = () => {
+  return new AppError({
+    code: 400,
+    error: 'Bad Request',
+    errno: ERRNO.UNVERIFIED_PRIMARY_EMAIL_NEWLY_CREATED,
+    message: 'Email already exists'
```

Comment:
Oh, wait, the messages are intentionally the same maybe? Profuse, fulsome apologies if so.

But I'd have thought we still want to differentiate the messages slightly though, it could be confusing to have the same message for different errnos?